core/src/main/java/com/graphhopper/routing/ch/PrepareContractionHierarchies.java

```diff
@@ -141,13 +158,12 @@ private void runGraphContraction() {
         } else {
             contractNodesUsingHeuristicNodeOrdering();
         }
-
     }
 
     @Override
     public RoutingAlgorithm createAlgo(Graph graph, AlgorithmOptions opts) {
         AbstractBidirAlgo algo = doCreateAlgo(graph, opts);
-        algo.setEdgeFilter(new LevelEdgeFilter(prepareGraph));
+        algo.setEdgeFilter(new LevelEdgeFilter(chGraph));
```

Comment:
Why do we still need the LevelEdgeFilter?

Reply:
We need it here for the same reason we always needed it: To filter out shortcuts going to/coming from lower level nodes. This PR does not change anything related to the CH algorithms (query algorithms) and here we just build the CH algorithm which uses the `LevelEdgeFilter`. Its indeed different for the CH preparation: We do not / no longer use LevelEdgeFilter for the preparation.

Reply:
Ah, missed that. Thought it had something to do with the edge vs. node-based difference.